### ISSUE TITLE
ERROR - Incorrect fingerprint for key_location

### ISSUE THREAD

**Issue by aaravindanarun** [jtriley/StarCluster#313]
I have installed the latest version of starcluster V0.9999 on Fedora 17 and stable version in v0.94 in Fedora 14. Install was success and now but launching the instance from Fedora 17.
Tried from same config and same key file from another fedora14 machine.
Its show some errors.  But when trying to launch the same config
I am getting the "!!! ERROR..." below. 

In Terminate :

> > > Using default cluster template: geusercluster1
> > > Validating cluster template settings...
> > > !!! ERROR - Cluster settings are not valid:
> > > !!! ERROR - Incorrect fingerprint for key_location
> > > !!! ERROR - '/home/Chandradeep/.ssh/genome_ap-southeast-1.pem'
> > > !!! ERROR - 
> > > !!! ERROR - local fingerprint:
> > > !!! ERROR - 1c:49:32:9c:00:93:52:f8:1c:78:1b:02:f1:99:7e:2d:74:16:da:7a
> > > !!! ERROR - 
> > > !!! ERROR - keypair fingerprint:
> > > !!! ERROR - 1c:95:23:ca:6e:c7:bc:d9:3e:45:2e:0c:b7:d3:96:e2:d6:2b:73:15

In Log File:

Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/StarCluster-0.94.2-py2.7.egg/starcluster/cli.py", line 275, in main
    sc.execute(args)
  File "/usr/lib/python2.7/site-packages/StarCluster-0.94.2-py2.7.egg/starcluster/commands/start.py", line 215, in execute
    validate_running=validate_running)
  File "/usr/lib/python2.7/site-packages/StarCluster-0.94.2-py2.7.egg/starcluster/cluster.py", line 1480, in start
    validator.validate()
  File "/usr/lib/python2.7/site-packages/StarCluster-0.94.2-py2.7.egg/starcluster/cluster.py", line 1669, in validate
    self.validate_keypair()
  File "/usr/lib/python2.7/site-packages/StarCluster-0.94.2-py2.7.egg/starcluster/cluster.py", line 2035, in validate_keypair
    % (key_location, keyfingerprint, fingerprint))
ClusterValidationError: Cluster settings are not valid:
Incorrect fingerprint for key_location '/home/Chandradeep/.ssh/genome_ap-southeast-1.pem'

local fingerprint: 1c:49:32:9c:00:93:52:f8:1c:78:1b:02:f1:99:7e:2d:74:16:da:7a

keypair fingerprint: 1c:95:23:ca:6e:c7:bc:d9:3e:45:2e:0c:b7:d3:96:e2:d6:2b:73:15

**Comment by jtriley** [comment]
@aaravindanarun That message means that the key you've specified does not match the fingerprint reported by EC2 for that keypair in that region.

Looking at the filename for your key it appears it was created in the ap-southeast-1 region.  Please verify that you're using the correct region either in your config or via the global --region option to the starcluster command. If you created separate keys for each region and they're all named the same then it's easy to mix them up and get this error.

**Comment by jtriley** [comment]
I'm going to close this because I believe this is a client configuration error. If you still think this is a bug please reopen.

**Comment by satra** [comment]
@jtriley - it seems this error happens on osx yosemite when the key is created with starcluster

```
$ starcluster -c config.sage createkey -o ~/xnat_configs/arno_key.pem arno_key 
StarCluster - (http://star.mit.edu/cluster) (v. 0.95.6)
Software Tools for Academics and Researchers (STAR)
Please submit bug reports to starcluster@mit.edu

>>> Successfully created keypair: arno_key
>>> fingerprint: 83:7e:0e:db:fa:85:f2:15:30:ed:5e:40:85:a0:bf:39:3a:41:e8:9c
>>> keypair written to /Users/satra/xnat_configs/arno_key.pem
(devpype)[satra@warpspeed starcluster]$ starcluster -c config.sage start -c smallcluster sagecluster --force-spot-master
StarCluster - (http://star.mit.edu/cluster) (v. 0.95.6)
Software Tools for Academics and Researchers (STAR)
Please submit bug reports to starcluster@mit.edu

>>> Validating cluster template settings...
!!! ERROR - Cluster settings are not valid:
!!! ERROR - Incorrect fingerprint for key_location
!!! ERROR - '/Users/satra/xnat_configs/arno_key.pem'
!!! ERROR - 
!!! ERROR - local fingerprint:
!!! ERROR - 83:7e:0e:db:fa:85:f2:15:30:ed:5e:40:85:a0:bf:39:3a:41:e8:9c
!!! ERROR - 
!!! ERROR - keypair fingerprint:
!!! ERROR - ba:a9:4b:24:63:1f:a8:1e:ca:a5:06:59:81:7c:15:59:c0:25:99:ae
```

**Comment by adeslatt** [comment]
So I am having this problem on OSX Yosemite as well -- did someone figure out how to fix it?  
I created a new keypair from amazon and from the pem file used 
sysgen-key to create the public/private rsa key for starcluster to use -- And I had the same problem and error -- which seems to say the key was made in the wrong region -- I check my config file and I do not specify the region.   So it would be helpful to know if this has been fixed.
